block-producer/src/state_view/mod.rs

```diff
@@ -44,10 +45,18 @@ where
     S: Store,
 {
     // TODO: Verify proof as well
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(skip_all, err(Debug), fields(COMPONENT = crate::COMPONENT))]
     async fn verify_tx(
         &self,
         candidate_tx: SharedProvenTx,
     ) -> Result<(), VerifyTxError> {
+        info!(
+            tx_id = %candidate_tx.id().inner(),
+            account_id = ?candidate_tx.account_id(),
+            COMPONENT,
+        );
```

Comment:
Can we not extract these from `candidate_tx` on line 49 above? This way, we can get rid of this extra line.

Also, maybe recording just the transaction ID here is enough?

Reply:
I've moved `account_id` to the `instrument`, since `tx_id` leaks from parent span. But maybe it would be better to avoid of fields in spans due to difficulties with such controlling them: they will leak to all subsequent calls and add verbosity.